src/main/kotlin/no/nav/helsearbeidsgiver/dokarkiv/DokArkivClient.kt

```diff
@@ -72,25 +71,15 @@ class DokArkivClient(
                     logger.error("Journalpost ble opprettet, men ikke ferdigstilt. journalpostId=[${it.journalpostId}] $idFragment")
                 }
             }
-            .onFailure {
-                if (it.instanceOf(ClientRequestException::class)) {
-                    it as ClientRequestException
-                    if (it.response.status == HttpStatusCode.Conflict) {
-                        val journalpost = runBlocking { it.response.body<OpprettOgFerdigstillResponse>() }
-                        if (!journalpost.journalpostId.isNullOrEmpty()) {
-                            logger.info(
-                                "Fikk 409 Conflict ved journalføring med referanse(id=${request.eksternReferanseId}) og tittel (${request.tittel}. " +
-                                    "Bruker eksisterende journalpostId (${journalpost.journalpostId}) fra respons.",
-                            )
-                            return journalpost
-                        } else {
-                            logger.error(
-                                "Fikk 409 Conflict ved journalføring med referanse(id=${request.eksternReferanseId}) og tittel (${request.tittel}, " +
-                                    "men mangler journalpostId fra respons.",
-                            )
-                        }
+            .recover {
+                val eksisterendeJournalpost =
+                    if (it is ClientRequestException && it.response.status == HttpStatusCode.Conflict) {
```

Comment:
`it is ClientRequestException` er en bedre måte å sjekke klassetype på, som automatisk caster ved suksess.